### DISCUSSION TITLE
Confuse about PBC

### DISCUSSION THREAD

**Discussion by saeedshahi** [mir-group/nequip#222] [Q&A] [answered]
Hello,
First, thank you for providing this excellent package for free.

I'm confused about how I should mention pbc in yaml input file. I'm going to use a water dataset in extxyz format made by ASE from a VASP OUTCAR.
If I use a standard extxyz file with keys like Lattice, energy, forces, pbc="T T T", so can NequIP match all keys automatically or I should mention them on the yaml input file?
What should I exactly write here:
dataset: ase                                                                    
dataset_file_name: Water64.extxyz                                    
 ase_args:
   format: extxyz
 include_keys:
   - user_label
 key_mapping:
   user_label: label0
espetioally include_keys and key_mapping.

Best,
Saeed

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @saeedshahi ,

When using ASE to load data, NequIP will take whatever PBC ASE is reading from the underlying data. There is no need to specify the PBC in the NequIP YAML; PBC are a per-frame property of the data and are thus always loaded from the data.

To confirm that your data is being recognized correctly as periodic by ASE, you can load your atoms directly using:
```
import ase.io
atoms = ase.io.read("whatever.xyz")
print(atoms.pbc)
```
and you could also visualize them if you like to confirm that the cell looks right.

If you have very non-standard keys in your XYZ for defining PBC, you may need to pre-process your XYZ file so that ASE can understand it. (`key_mapping` won't be enough to help you in that specific case.)

**Reply by saeedshahi** [reply]
Thanks for your prompt response. I obtain good RDF with NVT ensemble in MD by LAMMPS but when I use NPT, so my simulation box becomes bigger and bigger at each time step to reach P=1 atm. Does NequIP support NPT ensemble in LAMMPS?

**Reply by Linux-cpp-lisp** [reply]
This is discussed in a few other issues and discussion threads--- stress/NPT support in NequIP is still under development.

At present, training with stresses and using them with the ASE calculator is supported in beta on the latest main, v0.5.5. Support in LAMMPS `pair_nequip` is something we are working on.
